Landing page

SotirisFtiakas opened the pull request:
## About
This PR adds the landing page of the website. A landing page is the static page in which visitors "land" when entering the website, with the sole purpose of prompting them to take some action (e.g. register).
## Mockups vs Implementation 
Currently, the landing page consists of **three(3) parts.**
### Desktop

**First part.**

The main difference between the mockup and the implementation is a **white faded panel** behind the main elements to make them more pronounced, due to the **addition of a background image.** This background, as well as the following ones, have been implemented using an effect called **parallax**, where background images move slower than foreground images, creating an illusion of depth and adding to the sense of immersion in the virtual experience.
Other than that, the **scroll-down button** shown at the bottom is currently **not implemented**. It doesn't really add any additional functionality besides achieving a cool visual effect, so it's either going to be added in the distant future or might even be completely overlooked.
![image](https://user-images.githubusercontent.com/37048205/78985880-8b11c400-7b32-11ea-8376-4c2acf6a8166.png)
![image](https://user-images.githubusercontent.com/37048205/78985900-9664ef80-7b32-11ea-86a3-7131e69ba00a.png)


**Second part.**

Same difference as the first part. A faded white panel has been added in order to smoothly add a **background image frame** behind the cards. This background frame was added solely **for visual purposes.**
![image](https://user-images.githubusercontent.com/37048205/78985144-b2679180-7b30-11ea-859b-e82cf71dc6be.png)
![image](https://user-images.githubusercontent.com/37048205/78985280-08d4d000-7b31-11ea-9b84-4cf688216714.png)
 
**Third part.**

Same differences as the previous parts. 
![image](https://user-images.githubusercontent.com/37048205/78985356-36217e00-7b31-11ea-96af-6fba42e858c3.png)
![image](https://user-images.githubusercontent.com/37048205/78985376-40437c80-7b31-11ea-9244-b1e75fe4f54c.png)
 
### Mobile

**First part.**

![image](https://user-images.githubusercontent.com/37048205/78985937-bbf1f900-7b32-11ea-9512-c569d40edbac.png)

**Second part.**

![image](https://user-images.githubusercontent.com/37048205/78985956-c90ee800-7b32-11ea-9ffc-21897333123c.png)

**Third part.**

![image](https://user-images.githubusercontent.com/37048205/78985964-cf04c900-7b32-11ea-9957-ebe7f0b017b9.png)

## Responsiveness
The landing page is **fully responsive** and can support desktop, tablet and mobile adaptation. Changing the window size of the browser will adjust the content as needed, both horizontally and vertically.
 Moreover, for better user experience, tablets and mobiles will show only four(4) and three(3) event categories accordingly, to prevent information overload. This will also be dynamically applied to the desktop version, if the browser window's width gets smaller/bigger. 
## Possible changes
Things that might need to change are:
- **Font-type** and **font-sizes.** Due to double responsiveness (horizontally + vertically), few sentences adjust somewhat improperly in some cases.
- **Images.** We need to find some nice non-copyrighted images to make the landing page eye-catching.
- **Measurement types.** To make the landing page fully responsive, I used mostly vh and vw, only few rem and hardly any px. These can change depending on our judgement.
## Notices
- The translation and description of the event categories has not been implemented yet, as they will be extracted from the database. They are currently hardcoded using a json object.
- Due to the font non-responsiveness of the navigation-bar, there is significant difference when changing screen size, as the nav-bar letters are disproportionally bigger than the rest of the landing page. We might need to make the nav-bar font responsive or figure out some other solution.

GregB712 commented on .vscode/launch.json:
```diff
@@ -0,0 +1,17 @@
+            "program": "${workspaceFolder}\\bin\\dev.sh"
+        }
+    ]
+}
```
Redundant file.

din0s commented on src/www/components/landingpage/LandingPage.scss:
```diff
@@ -0,0 +1,612 @@
+.container {
+  background-image: linear-gradient(  rgb(129, 129, 129) ,white);
+  height: 325vh; //parallax(110vh) + 2 pages(200vh) + 15vh more empty space
```
You can just add a margin (top/bottom) instead of calculating manually how much space needs to be allocated.

```suggestion
  margin-bottom: 15vh;
```

din0s commented on src/www/components/landingpage/LandingPage.scss:
```diff
@@ -0,0 +1,612 @@
+.container {
+  background-image: linear-gradient(  rgb(129, 129, 129) ,white);
+  height: 325vh; //parallax(110vh) + 2 pages(200vh) + 15vh more empty space
+  max-width: 100vw;
```
Does this fix something? I didn't notice any differences when removing it locally.

din0s commented on src/www/components/landingpage/LandingPage.scss:
```diff
@@ -0,0 +1,612 @@
+  max-width: 100vw;
+  align-items: center;
+  justify-content: center;
+  font-family: 'Times New Roman', Times, serif;
```
We should find a font to use globally (apply in index.scss instead of per-container).

din0s commented on src/www/components/landingpage/LandingPage.scss:
```diff
@@ -0,0 +1,612 @@
+  .thirdpage{
+    
+    .phrase{
+      margin-top: 20vh;
```
I feel like this is too much vertical gap (same applies to all "header" elements).

din0s commented on src/www/components/landingpage/LandingPage.scss:
```diff
@@ -0,0 +1,612 @@
+      margin-bottom: 10vh;
+      float: left;
+      margin-left: 1  5vw;
+      font-size: calc(3.5vh + 1vw);
```
Why are you adjusting the font size based on the view-height?
I feel like the headers become too small when the height is limited.

din0s commented on src/www/components/landingpage/LandingPage.scss:
```diff
@@ -0,0 +1,612 @@
+      
+      width: 74vw;
+      height: 65vh;
+      margin-left: 13vw;
```
Why do we only need a left margin? (even though I couldn't find how this was being used anyway since the element's width is based on the view-width)

din0s commented on src/www/components/landingpage/LandingPage.scss:
```diff
@@ -0,0 +1,612 @@
+        flex-direction: row;
+        justify-content: center;
+        margin-left: 1rem;
+        margin-right: 1rem;
```
You can replace those two lines with:
```suggestion
        margin: 0 1rem;
```

din0s commented on src/www/components/landingpage/LandingPage.scss:
```diff
@@ -0,0 +1,612 @@
+          border-radius: 1rem;
+          background-color: rgba($color: white, $alpha: 0.7);
+          margin-left: 2rem;
+          min-width: 13rem;
```
Duplicate declaration, you have this rule already for this class.

din0s commented on src/www/components/landingpage/LandingPage.scss:
```diff
@@ -0,0 +1,612 @@
+          .boxdesc{
+            margin-top: 1vh;
+            margin-left: 1vh;
+            margin-right: 1vh;
```
I think we need a bottom margin for this one too, so replace these lines with
```suggestion
            margin: 1vh;
```

SotirisFtiakas commented:
**Update.** 
After my latest commit, here is some info on the requested changes:

**Done:**
•	Moved landing-page from components to containers
•	Removed unused image variable
•	Changed function to class
•	Removed typo
•	Sorted CSS rules alphabetically
•	Used "Prettier" formating
•	Re-added the needed navbar margin
•	Changed some translations
•	Gradient background is commented out for now.
•	Improved some margin syntax
•	Removed duplicate declarations

**To do:**
•	Form some common css rules
•	Remove unnecessary media queries (and use classes instead)
•	Change vertical gap on headers

SotirisFtiakas commented:
**Update.**
After my latest commit, the rest of the requested changes have been implemented:

#### Done: 
- Smaller header vertical spacing
- Removed media queries

#### Notices:
- I tried using only **vw** on font-sizes and other elements but the parallax effect collapses on this particular image. Since it doesn't affect the majority of the screen sizes, we can leave it as it is for now and when we decide on the images, it can be re-worked.
- The final background parallax images should preferably be vertically oriented.

din0s commented on src/www/containers/landingpage/LandingPage.js:
```diff
@@ -0,0 +1,173 @@
+          <p className="boxdesc">{categoriesList[key].desc}</p>
+        </div>
+      );
+    }
```
Add an else case with `return null` since we get the following warning:
```
Line 128:55:  Expected to return a value at the end of arrow function  array-callback-return
```
